### PR TITLE
reshaping and joins'

### PR DESCRIPTION
**Pull request by RohitRathore1**

### CODE REVIEW

**Comment by bkamins** [comment]
How does this PR relate to https://github.com/JuliaData/DataFrames.jl/pull/2717 as they seem to have duplicate content?

**Comment by RohitRathore1** [comment]
> How does this PR relate to #2717 as they seem to have duplicate content?

Yeah, Some content was useful. So I have used it.

**Comment by RohitRathore1** [comment]
You can review merge any PR because right now I have not added the file name in our make.jl file. When I will write all content about all functions then I will add those file names in make.jl. It will give me better picture in which sequence i should put those file name. And in the end if there will be any error then it will be resolved soon. There won't be too much difficulty.

**Review by pdeffebach — COMMENTED** [review]
Will continue reviewing later!

File: docs/src/man/joins_new.md [hunk 1]
```diff
@@ -0,0 +1,614 @@
...
+     │ String             Float64    String           Float64       Float64    ⋯
+─────┼──────────────────────────────────────────────────────────────────────────
+   1 │ biscuit                 0.05  caramel                  0.01        0.06 ⋯
+   2 │ biscuit                 0.05  chocolate sauce          0.05        0.1
+   3 │ chocolate biscuit       0.08  caramel                  0.01        0.09
+   4 │ chocolate biscuit       0.08  chocolate sauce          0.05        0.13
+   5 │ marshmallow             0.03  caramel                  0.01        0.04 ⋯
+   6 │ marshmallow             0.03  chocolate sauce          0.05        0.08
+```
+
```
[the remark this answers is not in the document]
Comment (RohitRathore1) [reply docs/src/man/joins_new.md:382]: Panel data?

[the remark this answers is not in the document]
Comment (bkamins) [reply docs/src/man/joins_new.md:382]: https://sites.google.com/site/econometricsacademy/econometrics-models/panel-data-models

File: docs/src/man/joins_new.md [hunk 2]
```diff
@@ -0,0 +1,614 @@
...
+The above four joins make up the basics of data frame merging. We can remember like this:
+- `innerjoin`: The output contains rows for values of the key that exist in both the first (left) and 
+  second (right) arguments to `join`.
+- `leftjoin`: The output contains rows for values of the key that exist in the first (left) argument to 
+  `join`, whether or not that value exists in the second (right) argument.
+- `rightjoin`: The output contains rows for values of the key that exist in the second (right) argument 
+  to `join`, whether or not that value exists in the first (left) argument.
+- `outerjoin`: The output contains rows for values of the key that exist in the first (left) or second 
+  (right) argument to `join`.
+
```
[the remark this answers is not in the document]
Comment (RohitRathore1) [reply docs/src/man/joins_new.md:265]: Regarding it, I have already given some definitions about those particular joins and then I have given examples about those joins respectively and then I have discussed for *Multiple keys to join on* for each joins. In this way, everything is in order for any user.

**Comment by RohitRathore1** [comment]
Not ready to review.

**Pull request closed without merging**